unit_tests/test_configuration.py

```diff
@@ -42,3 +91,19 @@ def test_migrate(self):
             fd.write("")
         self.configuration.migrate()
         self.assertThat(paths.LEGACY_BOOTSTRAP_FLAG, Not(FileExists()))
+
+    @contextlib.contextmanager
+    def _temporary_defaults_file(self):
+        """ Setup a temporary DEFAULTS_CONFIG_FILE
+        :return: a temporary file with the name attribute being the path
+        """
+        test_file_path = self.useFixture(fixtures.TempDir()).join("test-file")
```

Comment:
Thanks for using the TempDir fixture, so no temporary files are left around when the test fails.

However, I just realized that I could have suggested you a way simpler approach with systemfixtures:

http://systemfixtures.readthedocs.io/en/latest/?badge=latest#filesystem

that's the pattern that the rest of the unit test code in this charm uses. In your case I think you would have to write something like:

```python
self.fakes.fs.add(paths.DEFAULTS_CONFIG_FILE)
os.makedirs("/etc/default")
```
and from this point any call to open("/etc/default/jenkins") will be magically redirected to a temporary file (something like "/tmp/tmpxxx/etc/default/jenkins).

You could add the code above to unit_tests.states.AptInstallJenkins, and since ConfigurationTest already uses the AptInstalledJenkins state fixtures all you unit tests will have that default file already setup (since it's a file that is always expected to be there in a real unit).

I realize the above is a tad tricky, if you have troubles getting it done let me know, or just land this PR as it is and I'll take care of this small clean up myself.